addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/CacheableScanRule.java

```diff
@@ -99,11 +100,8 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
 
         // TODO: standardise the logic in the case of duplicate / conflicting headers.
         try {
-            if (logger.isDebugEnabled())
-                logger.debug(
-                        "Checking URL "
-                                + msg.getRequestHeader().getURI().getURI()
-                                + " for storability");
+            logger.debug(
+                    "Checking URL {} for storability", msg.getRequestHeader().getURI().getURI());
```

Comment:
```suggestion
                    "Checking URL {} for storability", msg.getRequestHeader().getURI().toString());
```
Just because it's more clear.

Reply:
Do you mean in escaped form? It's better not call `toString()` to not create the string sooner than needed (i.e. debug log is disabled).

Reply:
So what should I finally do @thc202 @kingthorin  ? Shall I keep `toString()` call everywhere or revert it?

Reply:
I think thc202 is saying it should simply be `msg.getRequestHeader().getURI()` and the JRE will toString it at runtime when needed.